File: web/src/api/piper.ts
import service from '@/utils/request';

// 获取Pipe列表
export const getPipeList = (params?: any) => {
    return service({
        url: '/api/v1/piper/list',
        method: 'get',
        params
    });
};

// 创建Pipe
export const createPipe = (data: any) => {
    return service({
        url: '/api/v1/piper/add',
        method: 'post',
        data
    });
};

// 删除Pipe
export const deletePipe = (data: any) => {
    return service({
        url: '/api/v1/piper/delete',
        method: 'post',
        data
    });
};

// 更新Pipe
export const updatePipe = (data: any) => {
    return service({
        url: '/api/v1/piper/update',
        method: 'post',
        data
    });
};

File: server/router/piper/piper.go
package piper

import (
	"gin-vue-admin/middleware"

	"github.com/gin-gonic/gin"
)

type PipeRouter struct{}

func (s *PipeRouter) InitPipeRouter(PrivateGroup *gin.RouterGroup) {
	apiRouter := PrivateGroup.Group("piper").Use(middleware.OperationRecord())
	apiWithNotRecorderRouter := PrivateGroup.Group("piper")
	{
		apiRouter.POST("add", pipeApi.AddPipe)       // 创建Pipe
		apiRouter.POST("delete", pipeApi.DeletePipe) // 删除Pipe
		apiRouter.POST("update", pipeApi.UpdatePipe) // 更新Pipe
	}
	{
		apiWithNotRecorderRouter.GET("list", pipeApi.GetPipeList) // 获取Pipe列表
	}
}
